src/components/drawing/index.js

```diff
@@ -482,6 +670,16 @@ drawing.singlePointStyle = function(d, sel, trace, fns, gd) {
             if(!gradientInfo[gradientType]) gradientType = 0;
         }
 
+        var getPatternAttr = function(mp, i, dflt) {
+            if(mp && Array.isArray(mp)) {
+                if(i < mp.length) return mp[i];
+                else return dflt;
```

Comment:
How about simply
```js
return i < mp.length ? mp[i] : dflt;
```

Reply:
Fixed in 51614783013bdc48ea9d8076f44a31a8da95b5a5